### PR TITLE
MODPERMS-83 Increase code coverage and fix code smells

### PR DESCRIPTION
**Pull request by adamdickmeiss**

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/PermsAPI.java [hunk 1]
```diff
@@ -382,15 +380,11 @@ public void deletePermsUsersById(String userid, Map<String, String> okapiHeaders
...
-      Criteria idCrit = new Criteria();
-      idCrit.addField(ID_FIELD);
-      idCrit.setOperation("=");
-      idCrit.setVal(userid);
-      PostgresClient pgClient = PostgresClient.getInstance(vertxContext.owner(),
-          tenantId);
+      Criterion idCrit = getIdCriterion(userid);
+      PostgresClient pgClient = PostgresClient.getInstance(vertxContext.owner(), tenantId);
       pgClient.startTx(connection -> {
         pgClient.get(TABLE_NAME_PERMSUSERS, PermissionUser.class,
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/PermsAPI.java:386]: should pass connection as first parameter

File: src/main/java/org/folio/rest/impl/PermsAPI.java [hunk 2]
```diff
@@ -428,7 +422,7 @@ public void deletePermsUsersById(String userid, Map<String, String> okapiHeaders
                     return;
                   }
                   pgClient.delete(connection, TABLE_NAME_PERMSUSERS,
-                      new Criterion(idCrit), deleteReply -> {
+                      idCrit, deleteReply -> {
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/PermsAPI.java:425]:
```suggestion
                      userid, deleteReply -> {
```

File: src/main/java/org/folio/rest/impl/PermsAPI.java [hunk 3]
```diff
@@ -857,12 +851,8 @@ public void getPermsPermissionsById(String id, Map<String, String> okapiHeaders,
                                       Context vertxContext) {
     try {
       String tenantId = TenantTool.tenantId(okapiHeaders);
-      Criteria idCrit = new Criteria();
-      idCrit.addField(ID_FIELD);
-      idCrit.setOperation("=");
-      idCrit.setVal(id);
       PostgresClient.getInstance(vertxContext.owner(), tenantId)
-          .get(TABLE_NAME_PERMS, Permission.class, new Criterion(idCrit), true, false, getReply -> {
+          .get(TABLE_NAME_PERMS, Permission.class, getIdCriterion(id), true, false, getReply -> {
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/PermsAPI.java:855]: Can PgUtil.getById been used?

File: src/main/java/org/folio/rest/impl/PermsAPI.java [hunk 4]
```diff
@@ -1063,7 +1046,7 @@ public void deletePermsPermissionsById(String id,
                         return;
                       }
                       pgClient.delete(connection, TABLE_NAME_PERMS,
-                          new Criterion(idCrit), deleteReply -> {
+                          idCrit, deleteReply -> {
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/PermsAPI.java:1049]:
```suggestion
                          id, deleteReply -> {
```

File: src/main/java/org/folio/rest/impl/PermsAPI.java [hunk 5]
```diff
@@ -1089,7 +1072,7 @@ public void deletePermsPermissionsById(String id,
             } else {
               try {
                 PostgresClient.getInstance(vertxContext.owner(), tenantId).delete(TABLE_NAME_PERMS,
-                    new Criterion(idCrit), deleteReply -> {
+                    idCrit, deleteReply -> {
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/PermsAPI.java:1075]:
```suggestion
                    id, deleteReply -> {
```

File: src/main/java/org/folio/rest/impl/PermsAPI.java [hunk 6]
```diff
@@ -1997,29 +1895,32 @@ private boolean allowAccessByPermission(String permissions, String permissionNam
...
       } else {
         return checkPermlistForDummy(permListCopy, vertxContext, tenantId);
       }
     });
   }
 
-  private Criteria getIdCriteria(String indexField, String operation, String value) {
+  private static Criterion getIdCriterion(String id) {
+    Criteria idCrit = new Criteria();
+    idCrit.addField(ID_FIELD);
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/PermsAPI.java:1917]: Add `idCrit.setJSONB(false)` to make use of the primary key index resulting in best performance:
https://github.com/folio-org/raml-module-builder/blob/master/domain-models-runtime/src/main/java/org/folio/rest/persist/Criteria/Criteria.java#L30

File: src/main/java/org/folio/rest/impl/PermsAPI.java [hunk 7]
```diff
@@ -1997,29 +1895,32 @@ private boolean allowAccessByPermission(String permissions, String permissionNam
...
+    idCrit.setOperation("=");
+    idCrit.setVal(id);
+    return new Criterion(idCrit);
+  }
+
+  private static Criterion getIdCriterion(String indexField, String value) {
     Criteria crit = new Criteria();
     if (indexField == null || indexField.equals("id")) {
       crit.addField(ID_FIELD);
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/PermsAPI.java:1926]: Add `idCrit.setJSONB(false)` to make use of the primary key index resulting in best performance:
https://github.com/folio-org/raml-module-builder/blob/master/domain-models-runtime/src/main/java/org/folio/rest/persist/Criteria/Criteria.java#L30

File: src/main/java/org/folio/rest/impl/TenantPermsAPI.java [hunk 8]
```diff
@@ -223,12 +175,12 @@ public void postTenantpermissions(OkapiPermissionSet entity, Map<String, String>
...
+      Iterator<Map.Entry<String, Future<Boolean>>> it = futureMap.entrySet().iterator();
       while (it.hasNext()) {
-        Map.Entry pair = (Map.Entry) it.next();
-        Future<Boolean> existsCheckFuture = (Future<Boolean>) pair.getValue();
-        if (existsCheckFuture.result() == false) {
-          notFoundList.add((String) pair.getKey());
+        Map.Entry<String, Future<Boolean>> pair = it.next();
+        Future<Boolean> existsCheckFuture = pair.getValue();
+        if (Boolean.FALSE.equals(existsCheckFuture.result())) {
+          notFoundList.add(pair.getKey());
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/TenantPermsAPI.java:183]:
```
      futureMap.forEach( (permName, existsCheckFuture) -> {
        if (Boolean.FALSE.equals(existsCheckFuture.result())) {
          notFoundList.add(permName);
        }
      });
```
or
```
      List<String> notFoundList = futureMap.entrySet().stream()
      .filter(entry -> Boolean.FALSE.equals(entry.getValue().result()))
      .map(Map.Entry::getKey)
      .collect(Collectors.toList());
```

File: src/main/java/org/folio/rest/impl/PermsAPI.java (hunk 1 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/rest/impl/PermsAPI.java]: Good catch

File: src/main/java/org/folio/rest/impl/PermsAPI.java (hunk 2 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/rest/impl/PermsAPI.java]: OK. Not that I think it improves readability.. idCrit is the one we are dealing with and when idCrit is using correct query type, it should be the same.

File: src/main/java/org/folio/rest/impl/PermsAPI.java (hunk 3 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/rest/impl/PermsAPI.java]: Actually, yes. But changes behavior a bit (not for the worse)

File: src/main/java/org/folio/rest/impl/PermsAPI.java (hunk 4 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/rest/impl/PermsAPI.java]: ok

File: src/main/java/org/folio/rest/impl/PermsAPI.java (hunk 5 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/rest/impl/PermsAPI.java]: ok

File: src/main/java/org/folio/rest/impl/PermsAPI.java (hunk 6 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/rest/impl/PermsAPI.java:1895]: Done and ID_FIELD must be unquoted.

File: src/main/java/org/folio/rest/impl/PermsAPI.java (hunk 7 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/rest/impl/PermsAPI.java:1905]: Yep

File: src/main/java/org/folio/rest/impl/TenantPermsAPI.java (hunk 8 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/rest/impl/TenantPermsAPI.java]: OK. Took the first one..

File: src/main/java/org/folio/rest/impl/PermsAPI.java [hunk 9]
```diff
@@ -117,10 +114,7 @@ private static CQLWrapper getCQL(String query, String tableName) throws FieldExc
 
   private final Messages messages = Messages.getInstance();
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/PermsAPI.java:115]: Can you remove this unused variable?

File: src/main/java/org/folio/rest/impl/PermsAPI.java [hunk 10]
```diff
@@ -117,10 +114,7 @@ private static CQLWrapper getCQL(String query, String tableName) throws FieldExc
...
 
-  private String getInternalError(String response) {
-    if (suppressErrorResponse) {
-      return "Internal Server Error: Please contact Admin";
-    }
+  static String getInternalError(String response) {
     return response;
   }
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/PermsAPI.java:119]: Can you remove this no-op method?

**Pull request merged**